src/main/java/com/github/manolo8/darkbot/gui/players/PlayerEditor.java

```diff
@@ -4,29 +4,86 @@
+import eu.darkbot.api.managers.EventBrokerAPI;
 import net.miginfocom.swing.MigLayout;
 
 import javax.swing.*;
-import java.awt.event.ActionEvent;
+import java.util.ArrayList;
+import java.util.Collections;
 import java.util.Comparator;
 import java.util.List;
 import java.util.Locale;
 
-public class PlayerEditor extends JPanel {
+public class PlayerEditor extends JPanel implements Listener {
     private final JList<PlayerInfo> playerInfoList;
     private final DefaultListModel<PlayerInfo> playersModel;
+    public static final List<PlayerInfo> nearbyPlayerList = new ArrayList<>();
 
     private final PlayerManager playerManager;
     private final SearchField sf;
     private final PlayerTagManager tagEditor;
     protected Main main;
 
+    public int indexOfPlayersModel(int playerId) {
+        for (int i = 0; i < playersModel.size(); i++)
+            if (playerId == playersModel.get(i).getUserId()) return i;
+        return -1;
+    }
+
+    @EventHandler
+    public void onEntityPlayerAdd(EntitiesAPI.EntityCreateEvent event) {
+        if (event.getEntity() instanceof Player
+                && playersModel != null
+                && main.config.PLAYER_INFOS != null
+                && !main.config.PLAYER_INFOS.containsKey(event.getEntity().getId())) {
+
+            SwingUtilities.invokeLater(() -> {
+                PlayerInfo playerInfo = new PlayerInfo((Player) event.getEntity());
+                if (indexOfPlayersModel(event.getEntity().getId()) == -1) {
+                    nearbyPlayerList.add(playerInfo);
+                    playersModel.addElement(playerInfo);
+                }
+            });
+        }
+    }
+
+    @EventHandler
+    public void onEntityPlayerRemove(EntitiesAPI.EntityRemoveEvent event) {
+        if (event.getEntity() instanceof Player
+                && playersModel != null
+                && main.config.PLAYER_INFOS != null
+                && !main.config.PLAYER_INFOS.containsKey(event.getEntity().getId())) {
+
+            SwingUtilities.invokeLater(() -> {
+                /*int idx = indexOf(event.getEntity().getId());
+                if (idx == -1 || playerInfoList.isSelectedIndex(idx)) return;
+                PlayerInfo info = playersModel.getElementAt(idx);
+                nearbyPlayerList.remove(info);
+                playersModel.removeElement(info);*/
+                for (int i = 0; i < playersModel.size(); i++) {
+                    PlayerInfo info = playersModel.getElementAt(i);
+                    if (info.getUserId() == event.getEntity().getId()) {
+                        if (!playerInfoList.isSelectedIndex(i)) {
+                            nearbyPlayerList.remove(info);
+                            playersModel.removeElement(info);
+                            break;
+                        } else {
+                            nearbyPlayerList.remove(info);
+                            break;
+                        }
+                    }
+                }
```

Comment:
What's wrong with the first (commented-out) version?
You shouldn't need to do a loop here, you have the indexOf version to find the element directly.

You will simply need to edit it a bit so that you have the logic of removing from nearbyPlayerList even if selected